.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,63 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# Sample workflow for building and deploying a Jekyll site to GitHub Pages
+name: Deploy Jekyll site to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: [$default-branch]
```

Comment:
@sneakers-the-rat i didn't know there was a $default-branch variable for actions. gosh that would have saved me a lot of time when we did the updates master --> main branch forever ago.

Reply:
Wait actually idk if this works lmao I think that might need to be a literal, but the sample action had this in it

Reply:
😆 so i'm not sure it does?? https://github.com/pyOpenSci/pyopensci.github.io/actions/workflows/gh-pages.yml 

when i look at that this [build](https://github.com/pyOpenSci/pyopensci.github.io/actions/workflows/gh-pages.yml) after directly commiting a small change to main 🙈 it wasn't triggered
<img width="359" alt="Screenshot 2024-07-23 at 4 49 21 PM" src="https://github.com/user-attachments/assets/f8b89b81-f5ef-42a0-8531-52c1534991e7">

but what did get triggered is [this action ](https://github.com/pyOpenSci/pyopensci.github.io/actions/workflows/pages/pages-build-deployment) 

could that be because i have github pages "turned on " in the repo settings somewhere? it's odd because the "pages" tab says this:
<img width="1071" alt="Screenshot 2024-07-23 at 4 50 45 PM" src="https://github.com/user-attachments/assets/33efcc63-722d-4bdd-bfcd-700d479f36b1">

Reply:
oh yes - sorry, so yeah that should be replaced by `main`, and then you go to the pages setting and switch it from "deploy from branch" to "github action" . we just did this switch for our lab website and after that it worked without fuss